### PR TITLE
Preference support for Guests

### LINKED ISSUE #72268: Guest accounts can't open the prefs menu
## Reproduction:

- Compile the game on a guest account for testing
- Open prefs menu
- Bluescreen error

I believe this is related to the 100s of json savefile related runtimes that Guest accounts also experience

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Makes it to where if you connect as a guest preferences don't error out the ass
Resolves https://github.com/tgstation/tgstation/issues/72268
## Why It's Good For The Game

Easier local debugging, and for servers that let guests join
## Changelog
:cl:
server: Preferences now support guests
/:cl:

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -2,20 +2,26 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 
 /datum/preferences
 	var/client/parent
-	//doohickeys for savefiles
+	/// The path to the general savefile for this datum
 	var/path
-	var/default_slot = 1 //Holder so it doesn't default to slot 1, rather the last one used
+	/// Whether or not we allow saving/loading. Used for guests, if they're enabled
+	var/load_and_save
```
Comment (Fikou) [inline code/modules/client/preferences.dm:8]: can you set a default value of true or false

Comment (ZephyrTFA) [reply code/modules/client/preferences.dm:8]: ah fuck ty

Comment (ZephyrTFA) [reply code/modules/client/preferences.dm:8]:
```suggestion
	var/load_and_save = TRUE
```

**Comment by Jacquerel** [comment]
Finally, this was a pain in the ass

**Review by Mothblocks — APPROVED** [review]

**Comment by Mothblocks** [comment]
let's TM this just in case

**Comment by Mothblocks** [comment]
well nboodys yelled at me

**Pull request merged**